test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -666,6 +667,33 @@ public void SerializationTest(string filePath)
                 serializationTestUtils.SaveWorkspaceComparisonData);
         }
 
+        /// <summary>
+        /// This parameterized test finds all .dyn files in directories within
+        /// the test directory, opens them and executes, then converts them to
+        /// json and executes again, comparing the values from the two runs
+        /// while being in a different culture.
+        /// </summary>
+        /// <param name="filePath">The path to a .dyn file. This parameter is supplied
+        /// by the test framework.</param>
+        [Test, TestCaseSource("FindWorkspaces"), Category("JsonTestExclude")]
+        public void SerializationTestInDifferentCulture(string filePath)
+        {
+            var frCulture = CultureInfo.CreateSpecificCulture("fr-FR");
+            
+            var currentCulture = Thread.CurrentThread.CurrentCulture;
+            var currentUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            Thread.CurrentThread.CurrentCulture = frCulture;
+            Thread.CurrentThread.CurrentUICulture = frCulture;
+
+            DoWorkspaceOpenAndCompare(filePath, jsonFolderName, ConvertCurrentWorkspaceToJsonAndSave,
+                serializationTestUtils.CompareWorkspaceModels,
+                serializationTestUtils.SaveWorkspaceComparisonData);
+
+            Thread.CurrentThread.CurrentCulture = currentCulture;
+            Thread.CurrentThread.CurrentUICulture = currentUICulture;
```

Comment:
does this actually set the culture back to the correct culture? could we assert it?

Reply:
I guess I meant does this actually set it back to en-us or something - I might be totally confused but this line `var currentCulture = Thread.CurrentThread.CurrentCulture;` 
 looks like it might just make a reference to the current Culture so when you set the culture to the french culture it seems that it would also update.

Is the currentCulture a value type or reference type?

Reply:
Yes, it sets it back to what was there before, in our case `en-us`. It must be value type because the switching works.